src/sentry/sentry_metrics/querying/data/postprocessing/remapping.py

```diff
@@ -0,0 +1,44 @@
+from collections.abc import Mapping, Sequence
+from typing import Any
+
+from sentry.models.project import Project
+from sentry.sentry_metrics.querying.data.execution import QueryResult
+from sentry.sentry_metrics.querying.data.mapping.mapper import Mapper
+from sentry.sentry_metrics.querying.data.postprocessing.base import PostProcessingStep
+
+
+class QueryRemappingStep(PostProcessingStep):
+    def __init__(self, projects: Sequence[Project]):
+        self.projects = projects
+
+    def run(self, query_results: list[QueryResult]) -> list[QueryResult]:
+        for query_result in query_results:
+            if query_result.totals is not None and len(query_result.totals) > 0:
+                query_result.totals = self._unmap_data(
+                    query_result.totals, query_result.totals_query.mappers
+                )
+            if query_result.series is not None and len(query_result.series) > 0:
+                query_result.series = self._unmap_data(
+                    query_result.series, query_result.series_query.mappers
+                )
+
+        return query_results
+
+    def _unmap_data(
+        self, data: Sequence[Mapping[str, Any]], mappers: list[Mapper]
+    ) -> Sequence[Mapping[str, Any]]:
+        for element in data:
+            updated_element = dict()
+            keys_to_delete = []
+            for result_key in element.keys():
+                for mapper in mappers:
+                    if mapper.to_key == result_key:
```

Comment:
How do we make sure that the mappers here are applied on the group by and not only on the filters?

Reply:
My question arises because if we have filtered by `project` and then we group by `project_id` (which can happen if someone were to send an API request on their own) we will map it back wrongly. We do not have to deal with this issue now but at least we know it's a possible problem. For now, the API is called mainly by the frontend so that we could rely on always have `project` but once the API will be public, this might bight us in the foot.